src/codewhisperer/util/dependencyGraph/goDependencyGraph.ts

```diff
@@ -0,0 +1,236 @@
+        for (const file of files) {
+            if (file.isDirectory() || !file.name.endsWith(DependencyGraphConstants.goExt) || file.name === fileName) {
+                continue
+            }
+            const filePath = path.join(packagePath, file.name)
+            const content = await readFileAsString(filePath)
+            if (this.readPackageName(content) !== packageName) {
+                continue
+            }
+            fileList.push(filePath)
+        }
+
+        return fileList
+    }
+
+    override async isTestFile(content: string): Promise<boolean> {
+        const imports = this.readImports(content)
+        const filteredImports = imports.filter(importStr => {
+            return importStr.includes('"testing"')
+        })
+        return filteredImports.length > 0
+    }
+
+    override async generateTruncation(uri: Uri): Promise<Truncation> {
+        try {
+            const workspaceFolder = vscode.workspace.getWorkspaceFolder(uri)
+            if (workspaceFolder === undefined) {
+                this._pickedSourceFiles.add(uri.fsPath)
+            } else {
+                await this.searchDependency(uri)
+                await this.traverseDir(this.getProjectPath(uri))
+            }
+            await sleep(1000)
+            getLogger().verbose('Picked source files: [' + [...this._pickedSourceFiles].join(', ') + ']')
+            const truncDirPath = this.getTruncDirPath(uri)
+            this.copyFilesToTmpDir(this._pickedSourceFiles, truncDirPath)
+            const zipFilePath = this.zipDir(truncDirPath, CodeWhispererConstants.codeScanZipExt)
+            const zipFileSize = statSync(zipFilePath).size
+            return {
+                rootDir: truncDirPath,
+                zipFilePath,
+                scannedFiles: new Set(this._pickedSourceFiles),
+                srcPayloadSizeInBytes: this._totalSize,
+                zipFileSizeInBytes: zipFileSize,
+                buildPayloadSizeInBytes: 0,
+                lines: this._totalLines,
+            }
+        } catch (error) {
+            getLogger().error('Go dependency graph error caused by:', error)
+            throw new Error('Go context processing failed.')
+        }
+    }
+
+    override async searchDependency(uri: Uri): Promise<Set<string>> {
+        const filePath = uri.fsPath
+        const q: string[] = []
+        q.push(filePath)
+        const siblings = await this.getSamePackageFiles(uri)
+        siblings.forEach(sibling => {
+            q.push(sibling)
+        })
+        while (q.length > 0) {
+            let count: number = q.length
+            while (count > 0) {
+                if (this.reachSizeLimit(this._totalSize)) {
+                    return this._pickedSourceFiles
+                }
+                count -= 1
+                const currentFilePath = q.shift()
+                if (currentFilePath === undefined) {
+                    throw new Error('"undefined" is invalid for queued file.')
```

Comment:
same comment about toolkit error

Reply:
Leaving this one as is because the previous one will catch this error